packages/trader/src/Modules/SmartChart/Components/Markers/accumulators-profit-loss-text.jsx

```diff
@@ -0,0 +1,134 @@
+        const interval_id = setInterval(() => {
+            if (action === ACTIONS.INC && profit_tenth_ref.current < end) {
+                profit_tenth_ref.current = (profit_tenth_ref.current + 1) % 10;
+            } else if (action === ACTIONS.DEC && profit_tenth_ref.current > end) {
+                profit_tenth_ref.current = Math.abs(profit_tenth_ref.current - 1) % 10;
+            } else if (action === ACTIONS.ADD10 && profit_tenth_ref.current < start + 10) {
+                profit_tenth_ref.current += 1;
+            } else if (
+                action === ACTIONS.ADD10 ? profit_tenth_ref.current === start + 10 : profit_tenth_ref.current === end
+            ) {
+                clearInterval(interval_id);
+            }
+            setCurrentProfitTenth(profit_tenth_ref.current % 10);
+        }, interval_ms);
+        interval_id_ref.current = interval_id;
+    };
+
+    React.useEffect(() => {
+        return () => {
+            clearTimeout(fading_in_timeout_id.current);
+            clearTimeout(sliding_timeout_id.current);
+            clearInterval(interval_id_ref.current);
+        };
+    }, []);
+
+    React.useEffect(() => {
+        if (profit) {
+            setIsFadingIn(true);
+            setIsSliding(true);
+            fading_in_timeout_id.current = setTimeout(() => {
+                setIsFadingIn(false);
+            }, 600);
+            sliding_timeout_id.current = setTimeout(() => {
+                setIsSliding(false);
+            }, 300);
+        }
+        if (profit !== 0) {
+            const updateTenth = (start, end) => {
+                const delta = Math.abs(end - start);
+                profit_tenth_ref.current = start;
+                if (start < end) {
+                    runThroughTenthDigit(ACTIONS.INC, 300 / delta, start, end);
+                } else if (start > end) {
+                    runThroughTenthDigit(ACTIONS.DEC, 300 / delta, start, end);
+                } else {
+                    runThroughTenthDigit(ACTIONS.ADD10, 30, start, end);
+                }
+            };
+            updateTenth(prev_profit_tenth, profit_tenth);
+        }
+    }, [profit, prev_profit_tenth, profit_tenth]);
+
+    const onRef = ref => {
+        if (ref) {
+            if (!current_spot) {
+                // this call will hide the marker:
+                ref.setPosition({ epoch: null, price: null });
+            }
+            ref.setPosition({
+                epoch: +current_spot_time,
+                price: +current_spot,
+            });
+        }
+    };
+
+    return (
+        <FastMarker markerRef={onRef} className={classNames(className, won ? 'won' : 'lost')}>
+            <Text
+                weight='bold'
+                size='m'
+                color={won ? 'profit-success' : 'loss-danger'}
+                className={classNames(`${className}__profit`, {
+                    [`${className}__profit--fading-in`]: is_fading_in,
+                })}
+                as='div'
```

Comment:
Just for my understanding why you used `as='div'` here?

Reply:
@niloo-fs I needed this when i applied `transform` css-property to this Text, which is `span` by default, because `transform` property doesn't apply to inline elements like `span` so I had to use it as a `div`. But I no longer need to use it as a `div` because I applied `display: flex` to this `span` which turned it into a block element. Good catch👍🏻 I'm gonna remove it